cmd/entrypoint-wrapper/main.go

```diff
@@ -326,6 +327,11 @@ func manageKubeconfig(proc *exec.Cmd) error {
 				if err := source.Close(); err != nil {
 					return true, err
 				}
+				newPath := strings.Trim(writableCopy.Name(), tempPrefix)
+				if err := os.Rename(writableCopy.Name(), newPath); err != nil {
+					return true, err
+				}
+				proc.Env = append(proc.Env, fmt.Sprintf("KUBECONFIG=%s", newPath))
```

Comment:
It seems that would require mutex set on proc and makes running rename inside a go func() a little bit complicated.

Reply:
I think we would like to avoid protecting fields right?